src/test/java/uk/nhs/hee/trainee/details/service/EventPublishServiceTest.java

```diff
@@ -60,4 +66,29 @@ void shouldPublishProfileCreateEvent() {
     ProfileCreateEvent event = eventCaptor.getValue();
     assertThat("Unexpected trainee ID.", event.getTraineeTisId(), is("10"));
   }
+
+  @Test
+  void shouldPublishCojSignedEvent() {
+    ProgrammeMembership programmeMembership = new ProgrammeMembership();
+    programmeMembership.setTisId("123");
+    Instant signedAt = Instant.now();
+    ConditionsOfJoining conditionsOfJoining
+        = new ConditionsOfJoining(signedAt, GoldGuideVersion.GG9);
+    programmeMembership.setConditionsOfJoining(conditionsOfJoining);
+
+    eventPublishService.publishCojSignedEvent(programmeMembership);
+
+    ArgumentCaptor<CojSignedEvent> eventCaptor = ArgumentCaptor.forClass(
+        CojSignedEvent.class);
+    verify(messagingTemplate).convertAndSend(eq(COJ_QUEUE_URL), eventCaptor.capture());
+
+    CojSignedEvent event = eventCaptor.getValue();
+    assertThat("Unexpected programme membership ID.",
+        event.getProgrammeMembershipTisId(), is("123"));
+    ConditionsOfJoining conditionsOfJoiningSent = event.getConditionsOfJoining();
+    assertThat("Unexpected CoJ Signed At",
+        conditionsOfJoining.signedAt(), is(conditionsOfJoiningSent.signedAt()));
+    assertThat("Unexpected CoJ Version",
+        conditionsOfJoining.version(), is(conditionsOfJoiningSent.version()));
```

Comment:
It's a bit of an edge case, but I've check against the original values rather than pulling them back out of `conditionsOfJoiningSent`. From a test perspective we can't guarantee a lack of side effects on the passed object tricking our test in to thinking all is well.

Reply:
Also, getting myself confused about which var was which highlighted that were testing these two var in an odd order. We're asserting on the original object rather than the captured "sent" object.

Reply:
(Sorry, response was to your original point. Yes, I picked up on the wrong order as well and fixed it in the next commit)